src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java

```diff
@@ -69,4 +100,233 @@ public boolean isExcluded(ProgrammeMembership programmeMembership) {
+  }
+
+  /**
+   * Remove a scheduled notification if it exists.
+   *
+   * @param jobId The job id key to remove.
+   * @throws SchedulerException if the scheduler failed in its duties (non-existent jobs do not
+   *                            trigger this exception).
+   */
+  private void removeNotification(String jobId) throws SchedulerException {
+    JobKey jobKey = new JobKey(jobId);
+    // Delete the job and unschedule its triggers.
+    // We do not simply remove the trigger, since a replacement job may have different data
+    // (e.g. programme name).
+    scheduler.deleteJob(jobKey);
+    log.info("Removed any stale notification scheduled for {}", jobId);
+  }
+
+  /**
+   * Get a future schedule for a notification from the programme start date and day offset.
+   *
+   * @param programmeStartDate The programme starting date.
+   * @param daysBeforeStart    The number of days prior to the programme start date.
+   * @return The notification scheduling date and time.
+   */
+  private Date getScheduleDate(LocalDate programmeStartDate, int daysBeforeStart) {
+    Date milestone;
+    LocalDate milestoneDate = programmeStartDate.minusDays(daysBeforeStart);
+    if (!milestoneDate.isAfter(LocalDate.now())) {
+      // 'Missed' milestones: schedule to be sent soon, but not immediately
+      // in case of human editing 'jitter'.
+      milestone = Date.from(Instant.now()
+          .plus(PAST_MILESTONE_SCHEDULE_DELAY_HOURS, ChronoUnit.HOURS));
+
+    } else {
+      // Future milestone.
+      milestone = Date.from(milestoneDate
+          .atStartOfDay()
+          .atZone(ZoneId.systemDefault())
+          .toInstant());
+    }
+    return milestone;
+  }
+
+  /**
+   * Helper function to determine whether a notification should be scheduled.
+   *
+   * @return true if it should be scheduled, false otherwise.
+   */
+  private boolean shouldScheduleNotification(LocalDate programmeStartDate,
+      NotificationType milestone, Map<NotificationType, Instant> notificationsAlreadySent) {
+
+    //do not resend any notification
+    if (notificationsAlreadySent.containsKey(milestone)) {
+      return false;
+    }
+
+    Integer daysBeforeStart = getNotificationDaysBeforeStart(milestone);
+    LocalDate milestoneDate = programmeStartDate.minusDays(daysBeforeStart);
+
+    if (milestoneDate.isAfter(LocalDate.now())) {
+      // A future notification, to be scheduled at the appropriate time
+      // unless a more recent notification has already been sent
+      return (notificationsAlreadySent.keySet().stream()
+          .noneMatch(t -> getNotificationDaysBeforeStart(t) < daysBeforeStart));
+    } else {
+      // A past ('missed') notification, to be scheduled promptly
+      // if it is the most recent missed notification.
+      // Do not schedule if it is not the most recent missed notification, regardless of whether
+      // any more recent ones have been sent or not.
+      return (getPastNotifications(programmeStartDate).stream()
```

Comment:
A/C 4:
Given that PGDiT has placement starting more then 8 weeks away
When the start date is changed to be within 8 & 4 weeks 
Then send the reminder email